src/tests.rs

```diff
@@ -778,6 +825,26 @@ fn parse_fn_prototype() {
     assert_debug_snapshot!(func);
 }
 
+#[test]
+fn parse_fn_mut_param() {
+    let func = parse_declaration(quote! {
+        fn prototype(a: i32, mut b: f32) -> String;
+    })
+    .unwrap();
+
+    assert_debug_snapshot!(func);
+}
```

Comment:
Minor nitpick: I'd rather these tests be a bit more consistent with existing tests:

- Use assert_quote_snapshot
- Use parse_declaration_checked
- Use `quote!` with parentheses
- Add a consume_generic_args_checked function that does the same thing as parse_declaration_checked

Though none of these are blockers, it's more for the code aesthetic ^^

Reply:
Done, except `assert_quote_snapshot`, here I wasn't sure what exactly you meant.

This is already part of `parse_declaration_checked`, no? `assert_debug_snapshot` seems more meaningful for the "data model" representation (and not the token one). Unless this is very important, I'd prefer if this one could be postponed, otherwise this PR is never-ending 🙂